platform/os_linux.go

```diff
@@ -115,14 +128,18 @@ func SetOutboundSNAT(subnet string) error {
 
 // ClearNetworkConfiguration clears the azure-vnet.json contents.
 // This will be called only when reboot is detected - This is windows specific
-func ClearNetworkConfiguration() (bool, error) {
+func (p *execClient) ClearNetworkConfiguration() (bool, error) {
 	return false, nil
 }
 
-func KillProcessByName(processName string) error {
-	p := NewExecClient()
+func (p *execClient) ExecutePowershellCommand(_ string) (string, error) {
+	return "", nil
+}
+
+func (p *execClient) KillProcessByName(processName string) error {
+	newExecClient := NewExecClient(p.logger)
 	cmd := fmt.Sprintf("pkill -f %v", processName)
-	_, err := p.ExecuteCommand(cmd)
+	_, err := newExecClient.ExecuteCommand(cmd)
```

Comment:
Why this change ? Shouldn't this be `p.ExecuteCommand` ?

Reply:
line 140 newExecClient := NewExecClient(p.logger)